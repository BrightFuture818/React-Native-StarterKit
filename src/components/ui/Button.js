/**
 * Buttons
 *
     <Button text={'Server is down'} />
 *
 * React Native Starter App
 * https://github.com/mcnamee/react-native-starter-app
 */
import React, { Component, PropTypes } from 'react';
import { Alert } from 'react-native';
import { Button } from 'react-native-elements';

// Consts and Libs
import { AppColors, AppFonts, AppSizes } from '@theme/';

/* Component ==================================================================== */
class CustomButton extends Component {
  static propTypes = {
    small: PropTypes.bool,
    large: PropTypes.bool,
    outlined: PropTypes.bool,
    backgroundColor: PropTypes.string,
    icon: PropTypes.shape({
      name: PropTypes.string,
    }),
  }

  static defaultProps = {
    small: false,
    large: false,
    outlined: false,
    icon: {},
  }

  buttonProps = () => {
    // Defaults
    const props = {
      title: 'Coming Soon...',
      onPress: () => Alert.alert('Coming soon...'),
      backgroundColor: AppColors.brand.primary,
      color: '#fff',
      fontWeight: 'bold',
      fontFamily: AppFonts.base.family,
      fontSize: AppFonts.base.size,
      borderRadius: AppSizes.borderRadius,
      raised: true,
      buttonStyle: {
        padding: 12,
        marginLeft: 0,
        marginRight: 0,
      },
      ...this.props,
      small: false,
      large: false,
      icon: (this.props.icon && this.props.icon.name)
        ? {
          ...this.props.icon,
          size: 14,
        } : null,
    };

    // Overrides
    // Size
    if (this.props.small) {
      props.fontSize = 12;
      props.buttonStyle.padding = 8;

      if (props.icon && props.icon.name) {
        props.icon = {
          ...props.icon,
          size: 14,
        };
      }
    }
    if (this.props.large) {
      props.fontSize = 20;
      props.buttonStyle.padding = 15;

      if (props.icon && props.icon.name) {
        props.icon = {
          ...props.icon,
          size: 20,
        };
      }
    }

    // Outlined
    if (this.props.outlined) {
      props.raised = false;
      props.backgroundColor = this.props.backgroundColor || 'transparent';
      props.color = AppColors.brand.primary;
      props.buttonStyle.borderWidth = 1;
      props.buttonStyle.borderColor = AppColors.brand.primary;

      if (props.icon && props.icon.name) {
        props.icon = {
          ...props.icon,
          color: AppColors.brand.primary,
        };
      }
    }

    return props;
  }

  render = () => <Button {...this.buttonProps()} />;
}

/* Export Component ==================================================================== */
export default CustomButton;
