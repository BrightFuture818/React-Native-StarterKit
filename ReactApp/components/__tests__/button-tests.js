/**
 * Test to check if the component renderes correctly
 */
'use strict'

import 'react-native'
import React from 'react'
import Button from '../button'
import renderer from 'react-test-renderer'

/**
 * Check if alerts renders correcly
 * and asserting it to the matching snapshot
 */
it('Button renders correcly', () => {

  const tree = renderer.create(
    <Button />
  ).toJSON()

  expect(tree).toMatchSnapshot()

});
