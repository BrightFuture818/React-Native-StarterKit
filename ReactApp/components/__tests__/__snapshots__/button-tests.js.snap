exports[`test Button renders correcly 1`] = `
<TouchableOpacity
  activeOpacity={0.7}
  disabled={false}
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#0069FF",
        "borderRadius": 3,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 10,
        "paddingHorizontal": 10,
      },
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#222222",
          "fontFamily": "Avenir",
          "fontSize": 14,
          "fontWeight": "500",
          "lineHeight": 21,
        },
        Object {
          "color": "#FFF",
          "fontFamily": "Avenir",
          "fontSize": 15,
          "fontWeight": "800",
          "lineHeight": 22,
          "textAlign": "center",
        },
      ]
    }>
    Click Here
  </Text>
</TouchableOpacity>
`;
